### ISSUE TITLE
Calculate correct layer for encoder in variational models

### ISSUE THREAD

**Issue by fdavidcl** [fdavidcl/ruta#33]
Seen in #32 that ruta's `encoder` does not have all necessary layers.

~~~r
x_train_mat <- matrix(1:200, nrow = 10)

network <-
  input() +
  dense(256, "elu") +
  variational_block(4, seed = 42, epsilon_std = .5) +
  dense(256, "elu") +
  output("sigmoid")

learner <- autoencoder_variational(network, loss = "binary_crossentropy")
model <- learner %>% train(x_train_mat, epochs = 5)

latent_model <- model$models$encoder

## Correct:
inputs <- keras::get_layer(model$models[[1]], index = 1)$input
latent_space <- keras::get_layer(model$models[[1]], index = 7)$output

latent_model <- keras::keras_model(
  inputs = inputs,
  outputs = latent_space
)
~~~
